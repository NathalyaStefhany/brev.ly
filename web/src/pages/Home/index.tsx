import React from 'react';

import { NewLink } from '@/pages/Home/components/NewLink';
import logo from '@/assets/logo.svg';

export const Home: React.FC = () => {
  return (
    <main className="min-h-screen flex flex-col items-center py-16 px-6 bg-gray-200">
      <div className="flex flex-col items-start gap-16">
        <img
          src={logo}
          alt="Ícone de corrente azul representando um link ao lado do nome 'brev.ly' também em azul."
          className="h-12"
        />

        <div className="flex flex-row flex-wrap justify-center gap-10">
          <NewLink />
        </div>
      </div>
    </main>
  );
};
