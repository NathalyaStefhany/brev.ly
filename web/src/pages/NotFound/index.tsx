import React from 'react';

import notFoundImage from '@/assets/404.svg';

export const NotFound: React.FC = () => {
  return (
    <main className="w-screen min-h-screen flex flex-row items-center justify-center p-6 bg-gray-200">
      <div className="max-w-290 w-fit h-fit flex flex-col gap-12 items-center text-center bg-gray-100 rounded-lg px-24 py-32">
        <img
          src={notFoundImage}
          alt="Número 404 representando o erro retornado ao acessar a página"
          className="h-42.5"
        />

        <h1 className="text-xl text-gray-600">Link não encontrado</h1>

        <p className="text-md text-gray-500">
          O link que você está tentando acessar não existe, foi removido ou é
          uma URL inválida. Saiba mais em{' '}
          <a href="/" className="text-blue-base underline">
            brev.ly
          </a>
          .
        </p>
      </div>
    </main>
  );
};
