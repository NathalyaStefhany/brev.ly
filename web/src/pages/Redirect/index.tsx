import React, { useEffect, useState } from 'react';

import { useLocation, useNavigate } from 'react-router';
import { api } from '@/service/api';
import logoIcon from '@/assets/logo-icon.svg';
import { AxiosError } from 'axios';
import { queryClient } from '@/service/queryClient';
import { refetchChannel } from '@/service/broadcastChannel';

type GetOriginalLinkOutput = {
  originalLink: string;
};

export const Redirect: React.FC = () => {
  const [rendered, setRendered] = useState(false);

  const { pathname } = useLocation();
  const navigate = useNavigate();

  useEffect(() => {
    const getOriginalLink = async (): Promise<void> => {
      try {
        const { data } = await api.get<GetOriginalLinkOutput>(
          `/shortened-links/${pathname.slice(1)}/original-link`,
        );

        let url = data.originalLink;

        if (url.startsWith('www.')) {
          url = `https://${url}`;
        }

        window.location.href = url;
      } catch (error) {
        const err = error as AxiosError;

        if (err.status === 404) {
          navigate('/url/not-found');
        }
      }
    };

    const updateAccessQuantity = async (): Promise<void> => {
      try {
        await api.patch(`/shortened-links/${pathname.slice(1)}/access`);

        queryClient.refetchQueries({ queryKey: ['links list'], exact: true });

        refetchChannel.postMessage({ queryKey: ['links list'], exact: true });
      } catch (error) {
        const err = error as AxiosError;

        if (err.status === 404) {
          navigate('/url/not-found');
        }
      }
    };

    if (pathname && rendered) {
      getOriginalLink();
      updateAccessQuantity();
    } else {
      setRendered(true);
    }
  }, [pathname, navigate, rendered]);

  return (
    <main
      className="w-screen min-h-screen flex flex-row items-center justify-center p-6 bg-gray-200"
      data-testid="container-redirect-page"
    >
      <div className="max-w-290 w-full h-fit flex flex-col gap-12 items-center text-center bg-gray-100 rounded-lg px-10 py-24 md:px-24 md:py-32">
        <img
          src={logoIcon}
          alt="Logotipo do site com um ícone de corrente azul que representa um link."
          className="h-24"
        />

        <h1 className="text-xl text-gray-600">Redirecionando...</h1>

        <div className="flex flex-col gap-2">
          <p className="text-md text-gray-500">
            O link será aberto automaticamente em alguns instantes.
          </p>

          <p className="text-md text-gray-500">
            Não foi redirecionado?{' '}
            <a
              href="/"
              className="text-blue-base underline"
              data-testid="link-access-here"
            >
              Acesse aqui
            </a>
          </p>
        </div>
      </div>
    </main>
  );
};
