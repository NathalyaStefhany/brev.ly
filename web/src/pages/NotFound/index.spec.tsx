import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { NotFound } from '@/pages/NotFound';

describe('Not found page tests', () => {
  it('should render correctly', () => {
    const { container } = render(<NotFound />);

    expect(container).toMatchSnapshot();
  });

  it('should go to home page when click on brev.ly link', () => {
    render(<NotFound />);

    expect(screen.getByTestId('link-brev-ly')).toHaveAttribute('href', '/');
  });
});
