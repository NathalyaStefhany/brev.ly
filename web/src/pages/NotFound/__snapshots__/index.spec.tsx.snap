// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Not found page tests > should render correctly 1`] = `
<div>
  <main
    class="w-screen min-h-screen flex flex-row items-center justify-center p-6 bg-gray-200"
  >
    <div
      class="max-w-290 w-fit h-fit flex flex-col gap-12 items-center text-center bg-gray-100 rounded-lg px-24 py-32"
    >
      <img
        alt="Número 404 representando o erro retornado ao acessar a página"
        class="h-42.5"
        src="/src/assets/404.svg"
      />
      <h1
        class="text-xl text-gray-600"
      >
        Link não encontrado
      </h1>
      <p
        class="text-md text-gray-500"
      >
        O link que você está tentando acessar não existe, foi removido ou é uma URL inválida. Saiba mais em
         
        <a
          class="text-blue-base underline"
          data-testid="link-brev-ly"
          href="/"
        >
          brev.ly
        </a>
        .
      </p>
    </div>
  </main>
</div>
`;
